app/models/components/course/model_component_host.rb

```diff
@@ -0,0 +1,33 @@
+class Course::ModelComponentHost
+  include Componentize
+
+  Course.after_create do
+    Course::ModelComponentHost.initialize_components(self)
+    true
+  end
+
+  def self.initialize_components(course)
+    components.each do |component|
+      component.course_initialize(course)
+    end
+  end
+
+  # Provides ::course_initialize to course components
+
+  module CourseComponentMethods
+    extend ActiveSupport::Concern
+
+    module ClassMethods
+      # A class method that course components may implement to hook into course initialisation.
+      #
+      # @param [Course] course The course under which the initialisation occurs.
```

Comment:
you should define a @!method so that yardoc knows what the course parameter maps to.